app/src/main/java/com/github/polypoly/app/base/game/location/InGameLocation.kt

```diff
@@ -47,4 +47,13 @@ data class InGameLocation (
     fun currentMortgagePrice(): Int {
         return locationProperty.baseMortgagePrice * (level.ordinal + 1)
     }
+
+    fun upgrade() {
+        level = when(level) {
+            PropertyLevel.LEVEL_0 -> PropertyLevel.LEVEL_1
+            PropertyLevel.LEVEL_1 -> PropertyLevel.LEVEL_2
+            PropertyLevel.LEVEL_2 -> PropertyLevel.LEVEL_MAX
+            PropertyLevel.LEVEL_MAX -> PropertyLevel.LEVEL_MAX
+        }
+    }
```

Comment:
You could use `values` and increment the array unless you're at the last index.

Reply:
Yes that's right, however I didn't do that as the code with `values()`, `valuesOf()` and the check that we are not in the last value would be a bigger and less clear code in the end. Moreover `when` doesn't compile if not exhaustive so it shouldn't have weird behavior anyway...